pkg-manager/core/src/install/index.ts

```diff
@@ -801,6 +802,19 @@ const _installInContext: InstallFunction = async (projects, ctx, opts) => {
     })
   }
 
+  // The wanted lockfile is mutated during installation. To compare changes, a
+  // deep copy before installation is needed. This copy should represent the
+  // original wanted lockfile on disk as close as possible.
+  //
+  // This object can be quite large. Intentionally avoiding an expensive copy
+  // if no lockfileCheck option was passed in.
+  const originalLockfileForCheck = opts.lockfileCheck != null
+    ? clone(ctx.wantedLockfile)
```

Comment:
It's a bit unfortunate that we have to clone the entire lockfile here. Since the `afterAllResolved` hook can change resolutions after installation, `pnpm dedupe --check` may report changes that the hook just reverts. Details in https://github.com/pnpm/pnpm/issues/6105#issuecomment-1484527964.

I went with an approach that diffs the lockfile rather than hooking into the `resolveDependencies` function.